Include/internal/pycore_unicodeobject.h

```diff
@@ -11,6 +11,230 @@ extern "C" {
+    const wchar_t *u,
+    Py_ssize_t size,
+    Py_ssize_t num_surrogates);
+extern PyObject* _PyUnicode_FromUCS2(const Py_UCS2 *u, Py_ssize_t size);
+extern PyObject* _PyUnicode_FromUCS4(const Py_UCS4 *u, Py_ssize_t size);
+extern PyObject* _PyUnicode_FromOrdinal(Py_UCS4 ordinal);
+extern PyObject* _PyUnicode_do_string_format(
+    PyObject *self,
+    PyObject *args,
+    PyObject *kwargs);
+extern PyObject* _PyUnicode_do_string_format_map(
+    PyObject *self,
+    PyObject *obj);
+extern Py_hash_t _PyUnicode_Hash(PyObject *self);
+extern PyObject* _PyUnicode_Iter(PyObject *seq);
+extern int _PyUnicode_IsModifiable(PyObject *unicode);
+extern void _PyUnicode_Fill(
+    int kind,
+    void *data,
+    Py_UCS4 value,
+    Py_ssize_t start,
+    Py_ssize_t length);
+extern PyObject* _PyUnicode_ResizeCompact(
+    PyObject *unicode,
+    Py_ssize_t length);
+extern int _PyUnicode_CheckModifiable(PyObject *unicode);
+extern PyObject* _PyUnicode_Repr(PyObject *unicode);
+extern PyObject* _PyUnicode_Pad(
+    PyObject *self,
+    Py_ssize_t left,
+    Py_ssize_t right,
+    Py_UCS4 fill);
+extern int _PyUnicode_Resize(PyObject **p_unicode, Py_ssize_t length);
+extern PyObject* _PyUnicode_EncodeUTF8(
+    PyObject *unicode,
+    _Py_error_handler error_handler,
+    const char *errors);
+extern PyObject* _PyUnicode_DecodeUTF8(
+    const char *s,
+    Py_ssize_t size,
+    _Py_error_handler error_handler,
+    const char *errors,
+    Py_ssize_t *consumed);
+extern char* _Py_backslashreplace(
+    PyBytesWriter *writer,
+    char *str,
+    PyObject *unicode,
+    Py_ssize_t collstart,
+    Py_ssize_t collend);
+extern char* _Py_xmlcharrefreplace(
+    PyBytesWriter *writer,
+    char *str,
+    PyObject *unicode,
+    Py_ssize_t collstart,
+    Py_ssize_t collend);
+extern PyObject* _PyUnicode_EncodeCallErrorHandler(
+    const char *errors,
+    PyObject **errorHandler,
+    const char *encoding,
+    const char *reason,
+    PyObject *unicode,
+    PyObject **exceptionObject,
+    Py_ssize_t startpos,
+    Py_ssize_t endpos,
+    Py_ssize_t *newpos);
+extern void _PyUnicode_RaiseEncodeException(
+    PyObject **exceptionObject,
+    const char *encoding,
+    PyObject *unicode,
+    Py_ssize_t startpos,
+    Py_ssize_t endpos,
+    const char *reason);
+extern int _PyUnicode_DecodeCallErrorHandlerWriter(
+    const char *errors,
+    PyObject **errorHandler,
+    const char *encoding,
+    const char *reason,
+    const char **input,
+    const char **inend,
+    Py_ssize_t *startinpos,
+    Py_ssize_t *endinpos,
+    PyObject **exceptionObject,
+    const char **inptr,
+    _PyUnicodeWriter *writer);
+extern PyObject* _PyUnicode_EncodeUCS1(
+    PyObject *unicode,
+    const char *errors,
+    const Py_UCS4 limit);
+
+extern PyTypeObject _Py_EncodingMapType;
+extern PyTypeObject _Py_FieldNameIter_Type;
+extern PyTypeObject _Py_FormatterIter_Type;
+
+/* helper macro to fixup start/end slice values */
+#define _Py_ADJUST_INDICES(start, end, len) \
```

Comment:
While we're at it, can we also have `start`, `end` and `len` with capital letters as asked by PEP-7?

Reply:
I would prefer to avoid reformatting in the middle of such huge surgery to help Git to follow the history.

Reply:
If you would like to apply a formatter, I suggest to do it in a separated PR. Latest discussion about that (in 2020): https://discuss.python.org/t/pep-proposal-automatically-formatting-the-cpython-code/5603